src/main/scala/com/dimafeng/testcontainers/TestContainersContext.scala

```diff
@@ -0,0 +1,51 @@
+package com.dimafeng.testcontainers
+
+import java.util.concurrent.CopyOnWriteArrayList
+
+import org.junit.runner.Description
+
+trait TestContainersContext {
+  def add(container: Container): Unit
+  def startAll()(implicit description: Description): Unit
+  def finishAll()(implicit description: Description): Unit
+  def failAll(e: Throwable)(implicit description: Description): Unit
+  def succeedAll()(implicit description: Description): Unit
+}
+
+object TestContainersContext {
+
+  class Empty extends TestContainersContext {
+    def add(container: Container): Unit = {}
+    def startAll()(implicit description: Description): Unit = {}
+    def finishAll()(implicit description: Description): Unit = {}
+    def failAll(e: Throwable)(implicit description: Description): Unit = {}
+    def succeedAll()(implicit description: Description): Unit = {}
+  }
+
+  class Default extends TestContainersContext {
+
+    import scala.collection.JavaConverters._
+
+    private val containers = new CopyOnWriteArrayList[Container]()
```

Comment:
Just thinking, maybe mutable scala collections would fit better (e.g. `mutable.ArrayBuffer`) to get rid of `asScala` conversions and have more scala-ish `add` - `+=`